service_kit/utils/timer.py

```diff
@@ -0,0 +1,64 @@
+import time
+from logging import Logger
+from typing import Any, Callable
+
+from service_kit.logging import logger as skit_logger
+
+
+class Timer:
+    """
+    Just another timer utility that can be used as a context manager or a decorator.
+
+    Example:
+       Context manager usage:
+         with Timer("some action"):
+             do_something()
+
+       Output:
+         "Timer finished", action="some action", time_taken_in_seconds=0.12345
+
+       Decorator usage:
+       @Timer("some function")
+       def some_function():
+         return do_something()
+
+       do_something()
+
+       Output:
+         "Timer finished", action="some function", time_taken_in_seconds=0.12345
+    """
+
+    def __init__(
+        self,
+        action: str,
+        get_time: Callable[[], float] = time.perf_counter,
+        logger: Logger = skit_logger,
```

Comment:
If you actually pass in a Python logger, does this work? What does the output look like? I suspect it won't include the `action` and `time_taken_in_seconds` fields. Maybe it will, I'm not sure.

If it doesn't work as expected, we should either change the definition or add a note to the docstring. If it only actually works with Loguru loggers, we should figure out the right type hint.

Reply:
I think we should stick to Loguru loggers for now. Fixed the type hint as per loguru [documentation](https://loguru.readthedocs.io/en/stable/api/type_hints.html).